commit 0a14231944a194b6bc31b737ff9b0dbec91edebc
Author: Sjoerd Simons <sjoerd@luon.net>
Date:   Sun Sep 17 20:25:25 2006 +0200

    Always set power_management.can_suspend to true if the power_management.type is
    pmu. Recent kernels don't report mem in /sys/power/state for ppc, but we have
    our own utility that issues ioctl's to suspend.

diff --git a/hald/linux/osspec.c b/hald/linux/osspec.c
index ed762a8..b9fb922 100644
--- a/hald/linux/osspec.c
+++ b/hald/linux/osspec.c
@@ -475,6 +475,11 @@ set_suspend_hibernate_keys (HalDevice *d
 		can_hibernate = TRUE;
 	free (poweroptions);
 
+	if (strcmp(hal_device_property_get_string(d, "power_management.type"), "pmu") == 0) {
+		/* We got our own helper for suspend PMU machines */
+		can_suspend = TRUE;
+	}
+
 	/* check for the presence of suspend2 */
 	if (access ("/proc/software_suspend", F_OK) == 0)
 		can_hibernate = TRUE;
